doc/user_guide/psyir.rst

```diff
@@ -45,6 +46,59 @@ source code using a frontend. Transformations act on the PSyIR and
 ultimately the generated code is produced by one of the PSyIR's
 backends.
 
+PSyIR Nodes
+-----------
+
+The PSyIR consists of classes whose instances can be connected
+together to form a tree which represent computation in a
+language-independent way. These classes all inherit from the ``Node``
+baseclass and, as a result, PSyIR instances are often referred to
+collectively as 'PSyIR nodes'.
+
+At the present time PSyIR classes can be essentially split into two
+types. PSy-layer classes and Kernel-layer classes. PSy-layer classes
+make use of a ``gen_code()`` or an ``update()`` method to create
+Fortran code whereas Kernel-layer classes make use of PSyIR backends
+to create code.
+
+.. note:: This separation will be removed in the future and eventually
+	  all PSyIR classes will make use of backends with the
+	  expectation that ``gen_code()`` and ``update()`` methods will
+	  be removed.
+
+PSy-layer nodes
+^^^^^^^^^^^^^^^
+
+PSy-layer PSyIR classes are primarily used to create the
+PSy-layer. These tend to be relatively descriptive and do not specify
+how a particular PSyclone frontend would implement them. With the
+exception of ``Loop``, these classes are currently not compatible with
+the PSyIR backends. The generic (non-api-specific) PSyIR nodes are:
```

Comment:
I think I've modified as suggested, but note that removing Invoke and replacing it with InvokeSchedule leads to a slight inconsistency as Kern and Directive are also not used in the PSy-layer, but I continue to include them, as they are subclassed into things that are, and they allow me to avoid writing out all of the subclasses in the case of directives, which would be a lot.

Reply:
Yes I know, but I think the main categorization that you are doing here are between PSy or Kernel nodes and some subclasses of schedule fall in different categories while Kern and Directive are always PSy. So I prefer the inconsistency in referring to abstract or specific classes than being unclear about if the are PSy or Kernel nodes.